### DISCUSSION TITLE
Where should domain events be triggerd?   Entity, or Command Handler?

### DISCUSSION THREAD

**Discussion by mweel1** [jasontaylordev/CleanArchitecture#1041] [Q&A]
Where should domain events live?

For example, if I have cancel appointment.

Appt.Cancel();

Should the cancel domain event happen in the Entity, or should it happen in the command handler that is receiving a cancellation request?

**Comment by TheMegaMatt** [comment]
In my opinion the CancelEvent should be raised by the entity after verifing that the event is cancellable, then you will have a EventHandler for said event.

**Reply by mweel1** [reply]
So I would inject mediator into the entity?

It feels weird injecting services into the entity, but maybe that's the way?

Thanks!

**Reply by devlear** [reply]
If your Entities implement BaseEntity, then you just need to add a new event to the Entity by calling AddDomainEvent.
Then when you save, the request will be intercepted by the SaveChangesInterceptor and routed to the matching EventHandler class
